src/Symfony/Bridge/Twig/Command/DebugCommand.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Lists twig functions, filters, globals and tests present in the current project
+ *
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ */
+class DebugCommand extends Command
+{
+    private $twig;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($name = 'debug:twig')
+    {
+        parent::__construct($name);
+    }
+
+    /**
+     * Sets the twig environment
+     *
+     * @param \Twig_Environment $twig
+     */
+    public function setTwigEnvironment(\Twig_Environment $twig)
+    {
+        $this->twig = $twig;
+    }
+
+    /**
+     * @return \Twig_Environment $twig
+     */
+    protected function getTwigEnvironment()
+    {
+        return $this->twig;
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDefinition(array(
+                new InputArgument('filter', InputArgument::OPTIONAL, 'Show details for all entries matching this filter'),
+                new InputOption('format', null, InputOption::VALUE_REQUIRED, 'Output format: text or json', 'text'),
+            ))
+            ->setDescription('Shows a list of twig functions, filters, globals and tests')
+            ->setHelp(<<<EOF
+The <info>%command.name%</info> command outputs a list of twig functions,
+filters, globals and tests. Output can be filtered with an optional argument.
+
+  <info>php %command.full_name%</info>
+
+The command lists all functions, filters, etc.
+
+  <info>php %command.full_name% date</info>
+
+The command lists everything that contains the word date.
+
+  <info>php %command.full_name% --format=json</info>
+
+The command lists everything in a machine readable json format.
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $twig = $this->getTwigEnvironment();
```

Comment:
The command should probably terminate with an error exit code if the Twig environment was not configured.